pkg/controller/certificates/signer/signer.go

```diff
@@ -138,22 +185,54 @@ func (s *signer) sign(x509cr *x509.CertificateRequest, usages []capi.KeyUsage) (
 	return pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: der}), nil
 }
 
-func requestValidForSignerName(req *x509.CertificateRequest, usages []capi.KeyUsage, signerName string) bool {
-	// Only handle CSRs with the specific known signerNames.
+// getCSRVerificationFuncForSignerName is a function that provides reliable mapping of signer names to verification so that
+// we don't have accidents with wiring at some later date.
+func getCSRVerificationFuncForSignerName(signerName string) (isRequestForSignerFunc, error) {
 	switch signerName {
 	case capi.KubeletServingSignerName:
-		return capihelper.IsKubeletServingCSR(req, usages)
+		return isKubeletServing, nil
 	case capi.KubeAPIServerClientKubeletSignerName:
-		return capihelper.IsKubeletClientCSR(req, usages)
+		return isKubeletClient, nil
 	case capi.KubeAPIServerClientSignerName:
-		return validAPIServerClientUsages(usages)
+		return isKubeAPIServerClient, nil
 	case capi.LegacyUnknownSignerName:
-		// No restrictions are applied to the legacy-unknown signerName to
-		// maintain backward compatibility in v1beta1.
-		return true
+		return isLegacyUnknown, nil
 	default:
+		// TODO type this error so that a different reporting loop (one without a signing cert), can mark
+		//  CSRs with unknown kube signers as terminal if we wish.  This largely depends on how tightly we want to control
+		//  our signerNames.
+		return nil, fmt.Errorf("unrecongized signerName: %q", signerName)
+	}
+}
+
+func isKubeletServing(req *x509.CertificateRequest, usages []capi.KeyUsage, signerName string) bool {
+	if signerName != capi.KubeletServingSignerName {
+		return false
+	}
+	return capihelper.IsKubeletServingCSR(req, usages)
```

Comment:
I would expect the signer to log an error if the signer name matches (and deny once https://github.com/kubernetes/enhancements/pull/1691 is implement), not ignore the CSR altogether.

Reply:
I see now that this is matches the existing behavior...

Reply:
> I see now that this matches the existing behavior...

Yes.  I'd be happy to see changes here once we can represent these failures (jordan's kep).  but I'd prefer to change behavior in another PR to make these easy to review.